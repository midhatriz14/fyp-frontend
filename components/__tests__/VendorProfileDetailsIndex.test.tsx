import { render, screen, waitFor } from "@testing-library/react-native";
import axios from "axios";
import React from "react";
import PhotographerDetailsScreen from "../vendorprofiledetails/VendorProfileDetailsIndex"; // Adjust path if needed

// Mocking axios for API calls with correct typing
jest.mock("axios");
const mockAxios = axios as jest.Mocked<typeof axios>;

// Mocking useGlobalSearchParams from expo-router correctly
jest.mock("expo-router", () => ({
  useGlobalSearchParams: jest.fn().mockReturnValue({ id: "123" }), // Mock the return value of useGlobalSearchParams
  router: {
    back: jest.fn(),
    push: jest.fn(),
  },
}));

describe("PhotographerDetailsScreen Component Tests", () => {
  it("should render loading indicator when data is being fetched", () => {
    render(<PhotographerDetailsScreen />);
    const loadingIndicator = screen.getByTestId("loading-indicator");
    expect(loadingIndicator).toBeTruthy();
  });

  it("should render an error message if vendor data is not available", async () => {
    // Mocking rejected API call to simulate failure
    mockAxios.get.mockRejectedValue(new Error("Failed to fetch"));

    render(<PhotographerDetailsScreen />);
    const errorMessage = await screen.findByTestId("error-message");
    expect(errorMessage).toHaveTextContent(
      "Failed to load vendor details. Please try again."
    );
  });

  it("displays error message when data fetching fails", async () => {
    mockAxios.get.mockRejectedValueOnce(new Error("Failed to fetch data"));
    render(<PhotographerDetailsScreen />);

    await waitFor(() => screen.getByTestId("error-message"));

    expect(screen.getByTestId("error-message")).toBeTruthy();
    expect(
      screen.getByText("Failed to load vendor details. Please try again.")
    ).toBeTruthy();
  });
});
