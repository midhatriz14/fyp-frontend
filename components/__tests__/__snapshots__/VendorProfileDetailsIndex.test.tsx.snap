// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotographerDetailsScreen Component Tests Snapshot-01: Matches snapshot for default state 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    color="#7B2869"
    size="large"
    testID="loading-indicator"
  />
  <Text>
    Loading vendor details...
  </Text>
</View>
`;

exports[`PhotographerDetailsScreen Component Tests Snapshot-02: Matches snapshot after data is loaded 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    color="#7B2869"
    size="large"
    testID="loading-indicator"
  />
  <Text>
    Loading vendor details...
  </Text>
</View>
`;

exports[`PhotographerDetailsScreen Component Tests Snapshot-03: Matches snapshot for packages tab 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    color="#7B2869"
    size="large"
    testID="loading-indicator"
  />
  <Text>
    Loading vendor details...
  </Text>
</View>
`;

exports[`PhotographerDetailsScreen Component Tests Snapshot-04: Matches snapshot for reviews tab 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    color="#7B2869"
    size="large"
    testID="loading-indicator"
  />
  <Text>
    Loading vendor details...
  </Text>
</View>
`;

exports[`PhotographerDetailsScreen Component Tests Snapshot-05: Matches snapshot when an error occurs 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    color="#7B2869"
    size="large"
    testID="loading-indicator"
  />
  <Text>
    Loading vendor details...
  </Text>
</View>
`;

exports[`PhotographerDetailsScreen Component Tests Snapshot-06: Matches snapshot for scroll view 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    color="#7B2869"
    size="large"
    testID="loading-indicator"
  />
  <Text>
    Loading vendor details...
  </Text>
</View>
`;

exports[`PhotographerDetailsScreen Component Tests Snapshot-07: Matches snapshot after switching to the reviews tab 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    color="#7B2869"
    size="large"
    testID="loading-indicator"
  />
  <Text>
    Loading vendor details...
  </Text>
</View>
`;
